execution/gethexec/espresso_finality_node.go

```diff
@@ -0,0 +1,158 @@
+package gethexec
+
+import (
+	"context"
+	"fmt"
+	espressoClient "github.com/EspressoSystems/espresso-sequencer-go/client"
+	espressoTypes "github.com/EspressoSystems/espresso-sequencer-go/types"
+	"github.com/ethereum/go-ethereum/arbitrum_types"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/arbos"
+	"time"
+
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbos/l1pricing"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+var (
+	retryTime = time.Second * 1
+)
+
+type HotsShotState struct {
```

Comment:
HotShotState

I am thinking if we really need this struct. It is simple

Reply:
makes sense, I will remove this!